subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskInputPropertiesIntegrationTest.groovy

```diff
@@ -417,31 +414,129 @@ apply from:'scriptPlugin.gradle'
             class MyTask extends DefaultTask {
                 @Input
                 FooType foo
+            }
+
+            task createFile(type: MyTask) {
+                ext.outputFile = file('output.txt')
+                outputs.file(outputFile)
+
+                foo = FooType.FOO
+
+                doLast {
+                    outputFile << foo
+                }
+            }
+"""
+
+        when:
+        succeeds 'createFile'
+
+        then:
+        executedTasks == [':createFile']
+        skippedTasks.empty
+
+        when:
+        succeeds 'createFile'
 
-                MyTask() {
-                    outputs.upToDateWhen {
-                        false
+        then:
+        executedTasks == [':createFile']
+        skippedTasks == [':createFile'] as Set
+    }
+
+    @Issue("gradle/gradle#919")
+    def "can use a custom Serializable type from init script as input property in this taskType: #taskType"() {
+        def initScript = file("init.gradle")
+
+        given:
+        initScript << """
+            enum FooType { FOO }
```

Comment:
We used to have an issue with enums for input properties (see https://issues.gradle.org/browse/GRADLE-3537 and https://issues.gradle.org/browse/GRADLE-3018). Even though these issues are fixed, @eriwen recently mentioned that we have some sort of issue lurking. Maybe he can elaborate on it.